docs/how-to-guides/performance_analysis.md

```diff
@@ -0,0 +1,206 @@
+# Performance analysis
+
+## Introduction
+
+Autoware is a real-time system, and it is important to have a small response time. If Autoware appears to be slow, it is imperative to conduct performance measurements and implement improvements based on the analysis. However, Autoware is a complex software system comprising numerous ROS 2 nodes, potentially complicating the process of identifying bottlenecks. To address this challenge, we will discuss methods for conducting detailed performance measurements for Autoware and provide case studies. It is worth noting that multiple factors can contribute to poor performance, such as scheduling and memory allocation in the OS layer, but our focus in this page will be on user code bottlenecks. The outline of this section is as follows:
+
+- Performance measurement
+  - Single node execution
+  - Prepare separated cores
+  - Run single node separately
+  - Measurement and visualization
+- Case studies
+  - Sensing component
+  - Planning component
+
+## Performance measurement
+
+Improvement is impossible without precise measurements.
+To measure the performance of the application code, it is essential to eliminate any external influences.
+Such influences include interference from the operating system and CPU frequency fluctuations.
+Scheduling effects also occur when core resources are shared by multiple threads.
+This section outlines a technique for accurately measuring the performance of the application code for a specific node.
+Though this section only discusses the case of Linux on Intel CPUs, similar considerations should be made in other environments.
+
+### Single node execution
+
+To eliminate the influence of scheduling, the node being measured should operate independently, using the same logic as when the entire Autoware system is running.
+To accomplish this, record all input topics of the node to be measured while the whole Autoware system is running.
+To achieve this objective, a tool called [`ros2_single_node_replayer`](https://github.com/sykwer/ros2_single_node_replayer) has been prepared.
+
+Details on how to use the tool can be found in the README.
+This tool records the input topics of a specific node during the entire Autoware operation and replays it in a single node with the same logic.
+
+### Prepare separated cores
+
+Isolated cores running the node to be measured must meet the following conditions.
+
+- Fix CPU frequency and disable turbo boost
+- Minimize timer interruptions
+- Offload RCU (Read Copy Update) callback
+- Isolate the paired core if hyper-threading enabled
+
+To fulfill these conditions on Linux, a custom kernel build with the following kernel configurations is required.
```

Comment:
I'm not Linux expert, but it might be difficult for people like me to modify a kernel configuration with given instruction in this page. Is possible to have more detailed explanation? Or are there any external tutorial where people could refer to?